src/npg_notify/porch_wrapper/qc_state.py

```diff
@@ -0,0 +1,151 @@
+      conf_file_path:
+        Configuration file path for this application.
+    """
+
+    porch_conf = get_config_data(
+        conf_file_path=conf_file_path,
+        conf_file_section=PORCH_CONFIG_FILE_SECTION,
+    )
+    # query LangQC for recently QC-ed wells
+    pipeline = Pipeline(
+        name=porch_conf["pipeline_name"],
+        uri=porch_conf["pipeline_uri"],
+        version=porch_conf["pipeline_version"],
+    )
+
+    langqc_conf = get_config_data(
+        conf_file_path=conf_file_path,
+        conf_file_section=LANGQC_CONFIG_FILE_SECTION,
+    )
+    qc_states = _send_request(
+        url=urljoin(langqc_conf["api_url"], langqc_conf["recently_qced_url"])
+    )
+
+    num_products = len(qc_states)
+    print(
+        f"Retrieved QC states for {num_products} products."
+    )  # TODO: log properly
+
+    os.environ["NPG_PORCH_TOKEN"] = porch_conf["npg_porch_token"]
+    del porch_conf["npg_porch_token"]
+
+    # qc_states is a dictionary where keys are product IDs and values are
+    # lists of QC states. A list of one QC state is expected.
+    num_errors = 0
+    for product_id, qc_state_data in qc_states.items():
+        try:
+            create_task(
+                porch_config=porch_conf,
+                pipeline=pipeline,
+                qc_state=qc_state_data[0],
+            )
+            # TODO: in DEBUG mode log the new task returned by this call.
+        except Exception as err:
+            # TODO: log the error message as an error
+            print(
+                f"Error registering QC state for product {product_id}: {str(err)}"
+            )
+            num_errors += 1
+
+    del os.environ["NPG_PORCH_TOKEN"]
+
+    print(
+        f"{num_errors} errors when registering products."
+        f"Registered QC states for {num_products-num_errors} products."
+    )  # TODO: log properly
+
+
+def create_task(
+    porch_config: dict[str], pipeline: Pipeline, qc_state: dict[str]
+):
+    """Creates and queues a single porch task for the notification pipeline.
+
+    Args:
+      porch_config:
+        A dictionary of porch-related configuration parameters.
+      pipeline:
+        npg_porch_cli.api.Pipeline object
+      qc_state:
+        A Python object that can be encoded into a JSON string. This object
+        uniquely defines a porch task for a given pipeline.
+    """
+
+    action = PorchAction(
+        validate_ca_cert=VERIFY_CERTIFICATE,
+        porch_url=porch_config["api_url"],
+        action="add_task",
+        task_input=qc_state,
+    )
+    send_porch_request(action=action, pipeline=pipeline)
+
+
+def _send_request(url: str):
+    headers = {
+        "Content-Type": "application/json",
+        "Accept": "application/json",
+    }
+    request_args = {
+        "headers": headers,
+        "timeout": CLIENT_TIMEOUT,
+        "verify": VERIFY_CERTIFICATE,
+    }
+
+    response = requests.request("GET", url, **request_args)
+    if not response.ok:
+        raise Exception(
```

Comment:
Maybe borrow `requests.HTTPError` for this?

Reply:
This code has been dropped.